src/export_json.lean

```diff
@@ -340,20 +340,52 @@ do docs ← olean_doc_strings,
      (filename, json.array $ l.map write_module_doc_pair) :: rest
    end) [])
 
-meta def get_instances : tactic (rb_lmap string string) :=
+/-- Extract `[foo, bar]` from `has_pow foo bar`.
+
+The result is a map containing the type names as keys.  -/
+meta def find_instance_types (e : expr) : tactic (rb_map string unit) :=
+list.mfoldl (λ m e, do
+  e ← dunfold [`decidable_rel, `decidable_pred] e {fail_if_unchanged := ff},
+  e ← whnf e transparency.reducible,
+  t ← infer_type e,
+  -- only look at arguments which are types or propositions
+  expr.sort l ← whnf t transparency.reducible | pure m,
```

Comment:
I'm not sure if the restriction is really necessary.

Reply:
I think this was motivated mainly by:

* Not having a good idea how to word the docs for the non-type cases
* Trying to limit the growth of the binary size of the doc

Reply:
I've removed this restriction, I realized that I wanted instances to appear for `lp`.